zipkin-server/src/main/java/zipkin2/server/internal/ZipkinHttpCollector.java

```diff
@@ -140,32 +141,40 @@ HttpResponse validateAndStoreSpans(SpanBytesDecoder decoder, ServiceRequestConte
         try {
           SpanBytesDecoderDetector.decoderForListMessage(nioBuffer);
         } catch (IllegalArgumentException e) {
-          result.onError(new IllegalArgumentException("Expected a " + decoder + " encoded list\n"));
+          future.completeExceptionally(new IllegalArgumentException("Expected a " + decoder + " encoded list\n"));
           return null;
         }
 
         SpanBytesDecoder unexpectedDecoder = testForUnexpectedFormat(decoder, nioBuffer);
         if (unexpectedDecoder != null) {
-          result.onError(new IllegalArgumentException(
+          future.completeExceptionally(new IllegalArgumentException(
             "Expected a " + decoder + " encoded list, but received: " + unexpectedDecoder + "\n"));
           return null;
         }
 
         List<Span> spans = new ArrayList<>();
         if (!decoder.decodeList(nioBuffer, spans)) {
-          result.onError(new IllegalArgumentException("Empty " + decoder.name() + " message"));
+          future.completeExceptionally(new IllegalArgumentException("Empty " + decoder.name() + " message"));
           return null;
         }
         // UnzippingBytesRequestConverter handles incrementing message and bytes
-        collector.accept(spans, result);
+        collector.accept(spans, new Callback<Void>() {
+          @Override public void onSuccess(Void value) {
+            future.complete(HttpResponse.of(ACCEPTED_RESPONSE));
+          }
+
+          @Override public void onError(Throwable t) {
+            future.completeExceptionally(t);
+          }
+        });
       } finally {
         ReferenceCountUtil.release(content);
       }
 
       return null;
     });
 
-    return HttpResponse.from(result);
+    return HttpResponse.from(future);
```

Comment:
I thought since I am wrapping the future with context aware executor, I wouldn't get a problem later.

However, it looks like maybe the real future is not wrapped for close event due to this stack trace:
```
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:1962) [?:1.8.0_181]
	at com.linecorp.armeria.common.HttpMessageAggregator.onComplete(HttpMessageAggregator.java:98) [armeria-0.88.0.jar:?]
	at com.linecorp.armeria.common.stream.AbstractStreamMessage$CloseEvent.notifySubscriber(AbstractStreamMessage.java:363) [armeria-0.88.0.jar:?]
	at com.linecorp.armeria.common.stream.DefaultStreamMessage.notifySubscriberOfCloseEvent(DefaultStreamMessage.java:194) [armeria-0.88.0.jar:?]
```